pkg/storage/imagestore/imagestore.go

```diff
@@ -933,7 +935,14 @@ func (is *ImageStore) FullBlobUpload(repo string, body io.Reader, dstDigest godi
 
 	uuid := u.String()
 	src := is.BlobUploadPath(repo, uuid)
-	digester := sha256.New()
+
+	var digester hash.Hash
+	if dstDigest.Algorithm() == godigest.SHA256 {
```

Comment:
we need to add support for SHA384 too:
```
else if dstDigest.Algorithm() == godigest.SHA384 {
   digester = sha512.New384()
}
```

Reply:
You should not need to reference sha256, sha384, or sha512 in your code.  The godigest library does that for you.

```go
digester := dstDigest.Algorithm().Hash() # hash.Hash is returned
```

Reply:
Of course you should check that the algorithm is available with `algDigest.Algorithm().Available()` (otherwise it will panic).

Reply:
The main branch of godigest even allows one to register new algorithms if they choose.  See [this](https://github.com/opencontainers/go-digest/blob/d50d2fec9c98606a94f0620bf09ed8a81a8aa237/algorithm.go#L78C11-L78C11).  So having the list of algorithms in Zot as a switch statement is not ideal.  Let the library from open containers handle it for you. :)